hazelcast-sql/src/main/java/com/hazelcast/jet/sql/impl/processors/StreamToStreamJoinP.java

```diff
@@ -325,22 +331,44 @@ public boolean saveToSnapshot() {
                 }
             }
 
-            Stream<Entry<?, ?>> leftBufferStream = buffer[0].content()
+            leftBufferStream = buffer[0].content()
                     .stream()
                     .map(row -> entry(
-                            ObjectArrayKey.project(row, joinInfo.leftEquiJoinIndices()),
+                            joinInfo.isEquiJoin() ? project(row, joinInfo.leftEquiJoinIndices()) : row.hashCode(),
```

Comment:
`JetSqlRow.hashCode()` returns constant `0`. Do we need unique value here?  hashCode is not good as unique value.